lsp/server/src/server.ts

```diff
@@ -201,87 +194,122 @@ async function dockerContainer(folder: WorkspaceFolder) {
 		});
 		startedSA4U_Z3 = true;
 	} catch (e) {
-		console.log(e);
+		connection.sendNotification('ServerError', `Failed to start the docker container: ${e}`);
 	}
 }
 
-connection.onShutdown(() => {
+async function changedConfiguration(change:DidChangeConfigurationParams) {
+	messDef = change.settings.SA4U.messageDefinition;
+	const readSA4U = (await promises.readFile(".sa4u.json")
+		.then((value)=> {return value.toString();})
+		.catch((err) => {console.warn(`Failed to read .sa4u.json: ${err}`); return new Boolean;}));
+	if (readSA4U instanceof Boolean) {
+		console.warn(`Failed to update .sa4u to the new message definition.`);
+		restartDockerContainer();
+		return;
+	}
+	const configJSON = JSON.parse(readSA4U);
+	configJSON['ProtocolDefinitionFile'] = messDef;
+	await promises.writeFile(".sa4u.json", JSON.stringify(configJSON, null, '    '))
+		.catch((err) => {console.warn(`Failed to update .sa4u file: ${err}`);});
+	restartDockerContainer();
+}
+
+async function restartDockerContainer(){
+	if (startedSA4U_Z3) {
+		startedSA4U_Z3 = false;
+		connection.workspace.getWorkspaceFolders().then((folders:null|WorkspaceFolder[]|void) => {
```

Comment:
I would just `await connection.workspace.getWorkspaceFolders()` since `restartDockerContainer` is already async

Reply:
Fixed